### ISSUE TITLE
Expand all locations when using Find All References

### ISSUE THREAD

**Issue by Hau-Hau** [microsoft/vscode#197592]
Hi,

There is no option to set "AlwaysExpand" for References view.

To achieve this goal it is needed to:
- Add new setting in ./extensions/references-view
- Read setting in every place that has hardcoded `vscode.TreeItemCollapsibleState.Collapsed`

![image](https://github.com/microsoft/vscode/assets/20733636/aea83f5b-b1b2-4eda-a088-6cb9fe49b6b9)

**Comment by jrieken** [comment]
Yeah, that's because we would need to load/open each file in order to expand it (we show a real textual preview) and that could be quite expensive

**Comment by Hau-Hau** [comment]
Is there anything specific what do you recommend to solve this problem?

My proposition is to handle it in same way as in Search view.
By that I mean to create one more option called "auto" which would expand files with less than 10 results.
So in result would be three options: auto, alwaysCollapse, alwaysExpand.

**Comment by ds614** [comment]
@Hau-Hau, could you please tell me how to view your PR? I am interested in trying your changes locally. Thanks!

**Comment by Hau-Hau** [comment]
Hi @ds614, PR you will find under: https://github.com/microsoft/vscode/pull/197594

Also, if you would like to hardcode "expanded" state without rebuilding vscode and all of that troubles you can directly override `extension.js`.
For Windows I'm using this command:
```powershell
(Get-Content "C:\Users\$($env:UserName)\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\references-view\dist\extension.js") -replace '.Collapsed', '.Expanded' | Set-Content "C:\Users\$($env:UserName)\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\references-view\dist\extension.js"
```

However, it is a bit annoying approach as after every vscode update, this file is regenerated to original form.

**Comment by ds614** [comment]
That's very helpful, thanks so much!

**Comment by ds614** [comment]
And by the way... I agree with you that I don't see what harm there would be in having this as a setting that defaults to the old "collapsed" behavior. 🤷‍♂️

**Comment by aguynamedben** [comment]
+1. The setting could be a configurable limit. For example, the option could called `referencesExpandResultsFileLimit` which defaults to `1` but it could be set to `10` if you're okay with the performance trade.

**Comment by tomerbarkan** [comment]
I implemented it locally and it really helps.
Any idea how to do this in Peek / Codelens view as well when finding references (C#)?

**Comment by Hau-Hau** [comment]
@tomerbarkan thanks!
I was never working around Peek functionality, so I cannot help you with that. However, if you would find way to do it, let me know!

@jrieken Any chances to move it forward? 
We have working PR + there was couple good ideas regarding potential performance issue.

**Comment by dslxvr** [comment]
It is worth noting that this feature is a **must** in order for "Find all implementations" search/filtering to be useful and work correctly. Please see the problems showcased on the attached GIF. If I want to find, for example, whether "Drop" is implemented for "Box", the search will NOT find it unless the result is expanded. The filtering also will not work until all results are expanded.

![Image](https://github.com/user-attachments/assets/9bdda840-9cff-425a-bb5d-b789bc49492c)

**Comment by niallobrien** [comment]
I'd really like to see this too
